pkg/workspace/library_module.go

```diff
@@ -92,12 +104,13 @@ func (l *libraryValue) WithDataValues(thread *starlark.Thread, f *starlark.Built
 
 	dataValues := core.NewStarlarkValue(args.Index(0)).AsGoValue()
 
-	libVal := &libraryValue{l.desc, l.libraryCtx, nil, l.libraryExecutionFactory}
+	libVal := &libraryValue{l.desc, l.libraryCtx, l.dataValuess, l.afterValuess, l.libraryDataValues, l.libraryExecutionFactory}
 	libVal.dataValuess = append([]*yamlmeta.Document{}, l.dataValuess...)
 	libVal.dataValuess = append(libVal.dataValuess, &yamlmeta.Document{
 		Value:    yamlmeta.NewASTFromInterface(dataValues),
 		Position: filepos.NewUnknownPosition(),
 	})
+	libVal.dataValuess = append(libVal.dataValuess, l.afterValuess...)
```

Comment:
dont think afterValuess should be appended here. they should be done inside eval() since one can write `library.get("foo").with_data_values(vals1).with_data_values(vals2)`